microceph/ceph/osd.go

```diff
@@ -264,25 +289,20 @@ func updateFailureDomain(s *state.State) error {
 	return nil
 }
 
-// AddOSD adds an OSD to the cluster, given a device path and a flag for wiping
-func AddOSD(s *state.State, path string, wipe bool, encrypt bool) error {
-	logger.Debugf("Adding OSD %s", path)
+// AddOSD adds an OSD to the cluster, given the data, WAL and DB devices and their respective
+// flags for wiping and encrypting.
+func AddOSD(s *state.State, data types.DiskParameter, wal *types.DiskParameter, db *types.DiskParameter) error {
+	logger.Debugf("Adding OSD %s", data.Path)
 
 	revert := revert.New()
 	defer revert.Fail()
 
 	// Validate the path.
-	if !shared.IsBlockdevPath(path) {
-		return fmt.Errorf("Invalid disk path: %s", path)
-	}
-	// Check if we need to support encryption
-	if encrypt {
-		if err := checkEncryptSupport(); err != nil {
-			return fmt.Errorf("Encryption unsupported on this machine: %w", err)
-		}
+	if !shared.IsBlockdevPath(data.Path) {
+		return fmt.Errorf("Invalid disk path: %s", data.Path)
 	}
 
-	_, _, major, minor, _, _, err := shared.GetFileStat(path)
+	_, _, major, minor, _, _, err := shared.GetFileStat(data.Path)
```

Comment:
We probably should look up a stable path for `wal.Path` and `db.Path` as well, like we do for `data.Path` here and further below. Might make sense to factor the lookup algo into a sep. function for this

Reply:
I didn't do it because Ceph itself will manage everything related to the wal and db paths (i.e: symlinking and all that), but yeah, it can be done.